tdrs-backend/tdpservice/search_indexes/management/commands/clean_and_reparse.py

```diff
@@ -263,32 +264,20 @@ def _handle_input(self, testing, continue_msg):
                 print('Cancelled.')
                 exit(0)
 
-    def handle(self, *args, **options):
-        """Delete and reparse datafiles matching a query."""
-        fiscal_year = options.get('fiscal_year', None)
-        fiscal_quarter = options.get('fiscal_quarter', None)
-        reparse_all = options.get('all', False)
-        new_indices = reparse_all is True
-
-        # Option that can only be specified by calling `handle` directly and passing it.
-        testing = options.get('testing', False)
-        ##
-
-        args_passed = fiscal_year is not None or fiscal_quarter is not None or reparse_all
-
-        if not args_passed:
-            logger.warn("No arguments supplied.")
-            self.print_help("manage.py", "clean_and_parse")
-            return
-
+    def get_files_to_reparse(case, fiscal_year, fiscal_quarter, selected_files, reparse_all):
+        """Get the files to reparse."""
         backup_file_name = "/tmp/reparsing_backup"
         files = DataFile.objects.all()
         continue_msg = "You have selected to reparse datafiles for FY {fy} and {q}. The reparsed files "
+        if selected_files:
+            files = files.filter(id__in=selected_files)
+            backup_file_name += "_selected_files"
+            continue_msg = continue_msg.format(fy=f"selected files: {str(selected_files)}", q="Q1-4")
         if reparse_all:
             backup_file_name += "_FY_All_Q1-4"
             continue_msg = continue_msg.format(fy="All", q="Q1-4")
         else:
-            if not fiscal_year and not fiscal_quarter:
+            if not fiscal_year and not fiscal_quarter and not selected_files:
```

Comment:
@raftmsohani related to the above, the model does not reflect that I selected fiscal year 2021 all quarters). how come?

![test1_model](https://github.com/user-attachments/assets/8b8e1e47-ba88-4d4b-9d5e-3346ec78cf86)

Reply:
from what i understand, this is expected. the `fiscal_quarter` and `fiscal_year` are populated when reparse is run via the management command. when run via the admin action, we only pass a list of the selected file ids, not any filter state (since that applies to the page as a whole, not necessarily which files are selected and included when the action is run)

Reply:
#3205 discusses refactoring the action so that it implements its own logic for reparse, separate from the management command. at that time we figured we could remove the management command altogether, and these fields on the reparse meta model as well

Reply:
this makes a lot of sense @jtimpe thank you! I'm going to re-test with a small set of files from DAC and from CLI while this command is still available, just to firm up my understanding how each method for reparsing affects the logs and model metadata.